### ISSUE TITLE
Deploy an Application: Switch to a logical cluster

### ISSUE THREAD

**Issue by pweil-** [kcp-dev/kcp#279]
Use Case: Switch to a logical cluster

Objectives:

- [x] User can create a workspace that will be considered as owned by him
- [x] User can view the list of the workspaces owned by him
- [x] User can delete a workspace owned by him
- [x] User can the list of the workspaces he has access to based on his group membership
- [x] User can switch to a workspace

This implies:
- [x] implementing a virtual workspace that provides the view on the personal workspaces of the user (or on all the workspaces of the organization his is part of and that he has permission to view). This endpoint would allow:
    - [x] getting / listing a workspace,
    - [x] creating / deleting a workspace,
    - [x] as well as retrieving the associated kubeconfig through a sub-resource.
```
kubectl use-context workspace-directory    # == some URL you can put into your .kube/config file that point to the virtual workspace
kubectl get workspaces                     # shows exactly your workspaces (those your have RBAC rights for)
```
- [x] implementing a `kubectl kcp workspace` plugin that builds upon the virtual workspace mentioned above and allows very easily working with workspaces while staying on a given kubectl context that points to the (kcp or not) cluster we want to work on. The plugin allows:
    - [x] Switching to a given workspace (by using the virtual workspace subresource adding a context to the current Kubeconfig)
    - [x] Switching back to the previous workspace (which implies storing the previous workspace)
    - [x] Getting the name of the current workspace
    - [x] Proxying the workspace list, create and delete actions (that could be done by kubectl) to make them easier while working on an arbitrary Kube context

About the `workspace` kubectl plugin, see https://docs.google.com/document/d/112nL8iitQZcVZQI8m-9nRo5alJL5PJFBiN6dfjhT-rA/edit#heading=h.y63dhu3wq36s

Note: A last feature in this area would be sharing workspaces: allowing a user to share (/ transfer ?) his personal workspace to another user. Will be added as a distinct issue.

**Comment by ncdc** [comment]
@davidfestal reminder from today's community meeting: please update the details section with what we plan to do for prototype 2. Thanks!

**Comment by davidfestal** [comment]
Current status:

- PR created and merged to implement the `workspaces` virtual workspace: #272 & #361
- Implementation of the sub-resource to retrieve the kubeconfig done, but not PR for now.
- Implementation in progress on the kubectl `workspace` plugin, but no PR for now.

**Comment by davidfestal** [comment]
Updated status:
- PR created for the `kubeconfig` sub-resource: https://github.com/kcp-dev/kcp/pull/375
- Kubectl `workspace` plugin implemented, but missing e2e tests (=> no PR for now)

**Comment by davidfestal** [comment]
- Kubectl workspace plugin implemented, with an associated demo with PR #376
